### PR TITLE
[ksqldb] Add implementation for describeCluster

### PR DESCRIPTION
**Pull request by eolivelli**
### Motivation

After upgrading the Kafka client library to 2.8.x some clients, like the KSQLDB server, require that describeCluster API is implemented.

### Modifications

- implement describeCluster
- response with UnsupportedVersionException to unknown APIs

### Verifying this change

This change added tests

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
No need to add docs, this is a standard API

- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java [hunk 1]
```diff
@@ -487,8 +491,17 @@ protected abstract void channelPrepare(ChannelHandlerContext ctx,
 
     protected abstract void completeCloseOnAuthenticationFailure();
 
-    protected abstract void
-    handleError(KafkaHeaderAndRequest kafkaHeaderAndRequest, CompletableFuture<AbstractResponse> response);
+    protected void handleError(KafkaHeaderAndRequest kafkaHeaderAndRequest,
```
[the remark this answers is not in the document]
Comment (Demogorgon314) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java:494]: If you want to move here, please also delete the `io.streamnative.pulsar.handlers.kop.KafkaRequestHandler#handleError` method.

[the remark this answers is not in the document]
Comment (BewareMyPower) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java:494]: @Demogorgon314 Is it a problem for you?

[the remark this answers is not in the document]
Comment (eolivelli) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java:494]: @Demogorgon314 I have deleted it
https://github.com/streamnative/kop/pull/1696/files#diff-f58b0ebcc57f58f5eaddb9cda8b023b15948068ec22b250650c73215c0b00279L496

**Comment by eolivelli** [comment]
This patch is also very important after the upgrade of the kafka client library

**Review by Demogorgon314 — APPROVED** [review]

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java [hunk 2]
```diff
@@ -2101,6 +2092,32 @@ protected void handleDescribeConfigs(KafkaHeaderAndRequest describeConfigs,
 
     }
 
+    @Override
+    protected void handleDescribeCluster(KafkaHeaderAndRequest describeConfigs,
+                                         CompletableFuture<AbstractResponse> resultFuture) {
+        checkArgument(describeConfigs.getRequest() instanceof DescribeClusterRequest);
+        DescribeClusterResponseData data = new DescribeClusterResponseData();
+        List<Node> allNodes = Collections.synchronizedList(
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java:2100]: Why do we use a synchronized list here?

Comment (eolivelli) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java]: good catch.
it was a left over of a copy/paste

**Review by Demogorgon314 — APPROVED** [review]

**Pull request merged**